### PR TITLE
Replace Contract.Requires with Debug.Assert

### PR DESCRIPTION
**Pull request by benaadams**
See:  #11596

### CODE REVIEW

File: src/System.Net.Http/src/System/Net/Http/HttpClient.cs [hunk 1]
```diff
@@ -535,7 +535,7 @@ private static void CheckBaseAddress(Uri baseAddress, string parameterName)
 
         private void SetTimeout(CancellationTokenSource cancellationTokenSource)
         {
-            Contract.Requires(cancellationTokenSource != null);
+            Debug.Assert(cancellationTokenSource != null);
```
Comment (stephentoub) [inline src/System.Net.Http/src/System/Net/Http/HttpClient.cs]: Can we remove the using imports for System.Diagnostics.Contracts in some of these files where Contract.Assert was the only use of contracts?

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStream.cs [hunk 2]
```diff
@@ -1171,7 +1171,7 @@ public override Task WriteAsync(Byte[] buffer, Int32 offset, Int32 count, Cancel
                                                         Task semaphoreLockTask, bool useApmPattern)
         {
 
-            // (These should be Contract.Requires(..) but that method had some issues in async methods; using Assert(..) for now.)
+            // (These should be Debug.Assert(..) but that method had some issues in async methods; using Assert(..) for now.)
```
Comment (stephentoub) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStream.cs]: These comments no longer make sense.

**Comment by stephentoub** [comment]
A few comments, but otherwise LGTM.  I admit I'm surprised to see the CI legs passing; I was expecting at least a few of the contracts would have been stale enough to cause a failure.

File: src/System.Net.Http/src/System/Net/Http/HttpClient.cs (hunk 1 above)
Comment (benaadams) [inline src/System.Net.Http/src/System/Net/Http/HttpClient.cs]: In my first pass I removed the using and project.json entries for them all; but it got a bit upset... As there were other uses.

Will make another pass.

**Comment by stephentoub** [comment]
Test Outerloop Windows_NT Debug please
Test Outerloop Ubuntu14.04 Debug please
Test Outerloop OSX Debug please

File: src/System.Net.Http/src/System/Net/Http/HttpClient.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Net.Http/src/System/Net/Http/HttpClient.cs]: Yeah, we can't for all of them, as there are other uses of Contract still, like Contract.Ensure, Contract.End*, etc.

Comment (benaadams) [inline src/System.Net.Http/src/System/Net/Http/HttpClient.cs]: Removed the usings and the references that aren't either needed or are brought in by dependencies. Should be more version consistency also.

**Comment by benaadams** [comment]
Removed the usings; tried to remove the unneeded references but its quite tricky - might need an automated process.

**Comment by stephentoub** [comment]
Thanks, Ben.

**Comment by stephentoub** [comment]
Test Outerloop Windows_NT Debug please
Test Outerloop OSX Debug please
Test Outerloop Ubuntu14.04 please

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStream.cs [hunk 3]
```diff
@@ -745,7 +745,7 @@ public override int Read([In, Out] Byte[] array, Int32 offset, Int32 count)
                                                             Task semaphoreLockTask, bool useApmPattern) {
 
         // Same conditions validated with exceptions in ReadAsync:
-        // (These should be Contract.Requires(..) but that method had some issues in async methods; using Assert(..) for now.)
+        // (These should be Debug.Assert(..) but that method had some issues in async methods; using Assert(..) for now.)
```
Comment (stephentoub) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStream.cs]: This comment no longer makes sense.

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStream.cs [hunk 4]
```diff
@@ -1170,8 +1170,6 @@ public override Task WriteAsync(Byte[] buffer, Int32 offset, Int32 count, Cancel
                                                         CancellationToken cancellationToken,
                                                         Task semaphoreLockTask, bool useApmPattern)
         {
-
-            // (These should be Contract.Requires(..) but that method had some issues in async methods; using Assert(..) for now.)
```
Comment (stephentoub) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStream.cs]: This comment can just be deleted.

**Comment by stephentoub** [comment]
Two remaining nits, otherwise LGTM.  Thanks, @benaadams.

**Pull request merged**